### PR TITLE
[FSDP][7/N] Add warning about frozen params

### PR DESCRIPTION
**Pull request by awgu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #104986
* #104969
* #104999
* __->__ #104967
* #104427

### CODE REVIEW

File: torch/distributed/fsdp/fully_sharded_data_parallel.py [hunk 1]
```diff
@@ -177,7 +177,18 @@ class FullyShardedDataParallel(nn.Module, _FSDPState):
...
+        ``param.requires_grad=False``). For ``use_orig_params=False``, each
+        FSDP instance must manage parameters that are all frozen or all
+        non-frozen. For ``use_orig_params=True``, FSDP supports mixing frozen
+        and non-frozen, but we recommend not doing so since then the gradient
+        memory usage will be higher than expected (namely, equivalent to not
+        freezing those parameters). This means that ideally, frozen parameters
+        should be isolated into their own ``nn.Module`` s and wrapped
+        separately with FSDP.
+
+    .. note::
```
Comment (awgu) [inline torch/distributed/fsdp/fully_sharded_data_parallel.py:191]: I do not think that the `::` matters compared to `:`, but I just unified it for consistency with the other warnings/notes.

Comment (Skylion007) [reply torch/distributed/fsdp/fully_sharded_data_parallel.py:191]: Ah, so this why you had that assert in FSDP.

Comment (awgu) [reply torch/distributed/fsdp/fully_sharded_data_parallel.py:191]: Haha is this in reference to the PR itself or this comment about `::` vs `:`?

I am guessing regarding the PR itself :)

Comment (Skylion007) [reply torch/distributed/fsdp/fully_sharded_data_parallel.py:191]: The PR itself

**Review by rohan-varma — APPROVED** [review]
LGTM

**Comment by awgu** [comment]
Failure looks unrelated:
```
ModuleNotFoundError: No module named 'torch.version'
```

**Pull request closed without merging**